misc/python/materialize/mzbuild.py

```diff
@@ -584,6 +592,7 @@ def build(self) -> None:
         for dep in self.dependencies.values():
             dep.acquire()
```

Comment:
`dep.acquire()` no longer does the right thing here, as acquire does not guarantee that the dependencies will exist. I think you should just remove this code, and document in the function's docstring that this function requires that the caller has already acquired all dependencies.